### PR TITLE
serial2TCP forwarding example

### PR DESCRIPTION
**Pull request by alexandrecuer**
with :
- signal to stop gracefully
- asyncio.run() to make it easier for people who don't know much about asyncio

### CODE REVIEW

**Comment by janiversen** [comment]
This looks good, I will review it soon. Would you like to add another PR that updates the general fowarder with your signal and async,run idea ?

**Comment by janiversen** [comment]
Please solve the pylint error.

**Comment by janiversen** [comment]
you can run the checks locally:
tox -e pylint
and
tox -e flake8

**Comment by janiversen** [comment]
please do not suppress pylint messages ! they are there for a reason.

**Comment by alexandrecuer** [comment]
I just left the `# pylint: disable=unused-argument` which is necessary for the graceful shutdown to work

if it is not acceptable, I will delete the pull request

**Comment by janiversen** [comment]
> I just left the `# pylint: disable=unused-argument` which is necessary for the graceful shutdown to work
> 
> if it is not acceptable, I will delete the pull request

I have not done a review yet I am primarily waiting to CI to give green light, and I am not sure why you threaten to delete the PR !

unused-argument can easily be avoided and I hardly believe the graceful shutdown depends on a pylint disable !

That being said unused-argument is something we accept if there are a good reason to have it.

File: examples/serial_forwarder.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        if self.server:
+            await self.server.shutdown()
+            _logger.info("TCP server is down")
+
+
+def get_commandline():
+    """Read and validate command line arguments"""
+    logchoices = ["critical", "error", "warning", "info", "debug"]
+
+    parser = argparse.ArgumentParser(description="Command line options for examples")
```
Comment (janiversen) [inline examples/serial_forwarder.py:64]: I think you mean "for serial_forwarder", at least these options are not valid for most examples.

File: examples/serial_forwarder.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    # set defaults
+    _logger.setLevel(args.log.upper() if args.log.lower() in logchoices else logging.INFO)
+    if not args.slaves:
+        args.slaves = {1, 2, 3}
+    return args.port, args.baudrate, args.server_port, args.server_ip, args.slaves
+
+
+if __name__ == "__main__":
+    server = SerialForwarderTCPServer()
+    import asyncio
```
Comment (janiversen) [inline examples/serial_forwarder.py:83]: Please move this to the top with the other imports.

**Review by janiversen — CHANGES_REQUESTED** [review]
The code looks quite good, but there are a couple of formal issues.

File: examples/serial_forwarder.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
+"""Pymodbus SerialRTU2TCP Forwarder
+
+usage :
+python3 serial_forwarder.py --log DEBUG --port "/dev/ttyUSB0" --baudrate 9600 --server_ip "192.168.1.27" --server_port 5020 --slaves 1 2 3
+
+sudo python3 serial_forwarder.py --port "/dev/ttyUSB0" --baudrate 9600 --server_ip "192.168.1.27" --server_port=502 --slaves 1 2 3
```
[the remark this answers is not in the document]
Comment (alexandrecuer) [reply examples/serial_forwarder.py:6]: To what i know, the default Modbus port number is 502

if you want to run on port 502, you need to use `sudo`

Ports below 1024 are restricted - only apps with root privileges can listen on those. I think this is the general rule on Linux/Unix

[the remark this answers is not in the document]
Comment (janiversen) [reply examples/serial_forwarder.py:6]: You are quite right depending on which system you are on. But independent you have already documented how to call it, so there are really no need to repeat it.

**Comment by alexandrecuer** [comment]
> unused-argument can easily be avoided and I hardly believe the graceful shutdown depends on a pylint disable !
> 
> That being said unused-argument is something we accept if there are a good reason to have it.

I dont know a better and more appropriate way to do it but if you know one, I am interested

I have a similar problem when using click in some projects. Pylint is not happy and I have to add `# pylint: disable=no-value-for-parameter`

**Review by janiversen — APPROVED** [review]

**Pull request merged**